FlowCal/mef.py

```diff
@@ -265,26 +319,30 @@ def fit_beads_autofluorescence(fl_channel, fl_mef):
     # Initial guesses:
     # 0: slope found by putting a line through the highest two points.
     # 1: y-intercept found by putting a line through highest two points.
-    # 2: bead autofluorescence initialized to 100.
+    # 2: bead autofluorescence initialized using the first point.
     params[0] = (np.log(fl_mef[-1]) - np.log(fl_mef[-2])) / \
-                    (fl_channel[-1] - fl_channel[-2])
-    params[1] = np.log(fl_mef[-1]) - params[0] * fl_channel[-1]
-    params[2] = 100.
+                (np.log(fl_au[-1]) - np.log(fl_au[-2]))
+    params[1] = np.log(fl_mef[-1]) - params[0] * np.log(fl_au[-1])
+    params[2] = np.exp(params[0]*np.log(fl_au[0]) + params[1]) - fl_mef[0]
 
     # Error function
     def err_fun(p, x, y):
-        return np.sum((np.log(y + p[2]) - ( p[0] * x + p[1] ))**2)
-        
+        return np.sum((np.log(y + p[2]) - ( p[0] * np.log(x) + p[1] ))**2)
+
     # Bead model function
     def fit_fun(p,x):
-        return np.exp(p[0] * x + p[1]) - p[2]
+        return np.exp(p[0] * np.log(x) + p[1]) - p[2]
 
-    # Channel-to-MEF standard curve transformation function
+    # a.u.-to-MEF standard curve transformation function
+    # We use the equivalent form ``exp(p[1]) * x**p[0]``. Mathematically, this
+    # is undefined for x < 0. In this case, we redefine the output as
+    # ``- exp(p[1]) * (-x)**p[0]``. These two can be combined as follows:
+    # ``np.sign(x) * np.exp(p[1]) * (np.abs(x)**p[0])``.
     def sc_fun(p,x):
-        return np.exp(p[0] * x + p[1])
+        return np.sign(x) * np.exp(p[1]) * (np.abs(x)**p[0])
```

Comment:
Might consider clarifying in the comment that `exp(p[1]) * x**p[0]` is only undefined for `x < 0` *and* non-integer `p[0]` (I think? perhaps a more precise set of numbers for `p[0]` for which that's undefined. It's still well defined for integer values of `p[0]`, though, I think, which is where I was getting hung up).

This all arises from this poorly justified `p[0]` slope parameter, which I still feel uneasy about. You're kinda arguing both sides here where you say it's consistently super close to 1 to justify this "extended" equation, but it's not consistently close enough to get rid of the parameter entirely.

I'm still trying to wrap my head around what exactly this is doing in the complex plane and how this would distort data if `p[0]` deviates more significantly from 1.0. Will come back to this...